docs/source/usage.rst

```diff
@@ -740,6 +740,22 @@ If resolve is set to True, interpolations will be resolved during conversion.
     >>> print(resolved)
     {'foo': 'bar', 'foo2': 'bar'}
 
+If you want OmegaConf.to_container to handle structured configs in a special
+way, you can set the `structured_config_mode` option:
+
+.. doctest::
+
+    >>> from omegaconf import SCMode
+    >>> conf = OmegaConf.create({"structured_config": MyConfig})
+    >>> container = OmegaConf.to_container(conf, structured_config_mode=SCMode.DICT_CONFIG)
+    >>> print(container)
+    {'structured_config': {'port': 80, 'host': 'localhost'}}
+    >>> assert type(container) == dict
+    >>> assert type(container["structured_config"]) == DictConfig
+
+Using `structured_config_mode=SCMode.DICT_CONFIG` means that the `to_container`
+function will not convert structured configs to primitive containers; they will
+remain as `DictConfig` instances.
```

Comment:
I suggest something like this.
You may need to change your initialization of the config to support the last test.
```
You can customize the treatment of **OmegaConf.to_container()** for Structured Config nodes using
the `structured_config_mode` option.
By default, Structured Config nodes are converted to plain dict. Using **structured_config_mode=SCMode.DICT_CONFIG** causes such nodes to remain DictConfig, allowing attribute style access on the resulting node.

.. doctest::

    >>> from omegaconf import SCMode
    >>> conf = OmegaConf.create({"structured_config": MyConfig})
    >>> container = OmegaConf.to_container(conf, structured_config_mode=SCMode.DICT_CONFIG)
    >>> print(container)
    {'structured_config': {'port': 80, 'host': 'localhost'}}
    >>> assert type(container) is dict
    >>> assert type(container["structured_config"]) is DictConfig
    >>> assert container["structured_config"].SOME_ATTRIBUTE_ACCESS == SOME_VALUE
```

Reply:
I revised the docs in https://github.com/omry/omegaconf/pull/552/commits/4dff6c1381edaf14b7aa01ebb1a95466279cf930.

Reply:
I've also updated the docstring for `OmegaConf.to_container`, in https://github.com/omry/omegaconf/pull/552/commits/c63587983e9bfeb7fac1b1dfa42128a4cfea3246.